simtools/db/db_handler.py

```diff
@@ -1208,10 +1208,19 @@ def get_all_versions(
 
         if _cache_key not in DatabaseHandler.model_versions_cached:
             if self._get_db_name():
-                db_collection = DatabaseHandler.db_client[self._get_db_name()][collection]
-                DatabaseHandler.model_versions_cached[_cache_key] = list(
-                    {post["version"] for post in db_collection.find(query)}
-                )
+                if collection is None:
+                    all_versions = set()
+                    for collection_name in self.get_collections(self._get_db_name(), True):
+                        db_collection = DatabaseHandler.db_client[self._get_db_name()][
+                            collection_name
+                        ]
+                        all_versions.update(post["version"] for post in db_collection.find(query))
+                    DatabaseHandler.model_versions_cached[_cache_key] = list(all_versions)
+                else:
+                    db_collection = DatabaseHandler.db_client[self._get_db_name()][collection]
+                    DatabaseHandler.model_versions_cached[_cache_key] = list(
+                        {post["version"] for post in db_collection.find(query)}
+                    )
```

Comment:
These lines became a bit hard to read, but unfortunately I cannot come up with an idea on how to improve it. Perhaps `_db_name = self._get_db_name()` and then using `_db_name` everywhere would at least make the lines shorter, but it doesn't help with the series of ifs and for loops.

Reply:
Had another look and simplified it significantly.

Reply:
Looks much better thanks!

I am a bit surprised none of the linters complained about `if collection`. I thought they require to check explicitly `if collection is not None`. I don't mind it personally, so if it passes, it's fine.